geth/transactions/txqueue_manager_test.go

```diff
@@ -279,3 +280,62 @@ func (s *TxQueueTestSuite) TestCompletionTimedOut() {
 	rst := txQueueManager.WaitForTransaction(tx)
 	s.Equal(ErrQueuedTxTimedOut, rst.Error)
 }
+
+// TestLocalNonce verifies that local nonce will be used unless
+// upstream nonce is updated and higher than a local
+// in test we will run 3 transaction with nonce zero returned by upstream
+// node, after each call local nonce will be incremented
+// then, we return higher nonce, as if another node was used to send 2 transactions
+// upstream nonce will be equal to 5, we update our local counter to 5+1
+func (s *TxQueueTestSuite) TestLocalNone() {
```

Comment:
That looks good! I would also add a fail case where it is verified that nonce was not incremented.